blas/unit_test/Test_Blas2_syr.hpp

```diff
@@ -0,0 +1,1760 @@
+      << "Failed test, '" << situation
+      << "': kk ger() should not have thrown a std::exception";
+
+  // ********************************************************************
+  // Prepare h_ger_reference to be compared against h_A_syr
+  // ********************************************************************
+  view_stride_adapter<_ViewTypeExpected, true> h_ger_reference(
+      "h_ger_reference", _M, _N);
+  Kokkos::deep_copy(h_ger_reference.d_base, A_ger.d_base);
+
+  std::string uplo = _useUpOption ? "U" : "L";
+  for (int i = 0; i < _M; ++i) {
+    for (int j = 0; j < _N; ++j) {
+      if (((_useUpOption == true) && (i <= j)) ||
+          ((_useUpOption == false) && (i >= j))) {
+        // Keep h_ger_reference as already computed
+      } else {
+        h_ger_reference.d_view(i, j) = org_A.h_view(i, j);
+      }
+    }
+  }
+  if (_useHermitianOption && _A_is_complex) {
+    for (int i(0); i < _N; ++i) {
+      h_ger_reference.d_view(i, i) =
+          0.5 * (h_ger_reference.d_view(i, i) +
+                 _KAT_A::conj(h_ger_reference.d_view(i, i)));
+    }
+  }
+
+  // ********************************************************************
+  // Compare
+  // ********************************************************************
+  this->compareKkSyrAgainstReference(alpha, h_A_syr, h_ger_reference.d_view);
+}
+
+}  // namespace Test
+
+template <class ScalarX, class ScalarA, class Device>
+int test_syr(const std::string& caseName) {
+  KOKKOS_IMPL_DO_NOT_USE_PRINTF(
+      "+======================================================================="
+      "===\n");
+  KOKKOS_IMPL_DO_NOT_USE_PRINTF("Starting %s ...\n", caseName.c_str());
+
+  bool xBool = std::is_same<ScalarX, float>::value ||
+               std::is_same<ScalarX, double>::value ||
+               std::is_same<ScalarX, Kokkos::complex<float>>::value ||
+               std::is_same<ScalarX, Kokkos::complex<double>>::value;
+  bool aBool = std::is_same<ScalarA, float>::value ||
+               std::is_same<ScalarA, double>::value ||
+               std::is_same<ScalarA, Kokkos::complex<float>>::value ||
+               std::is_same<ScalarA, Kokkos::complex<double>>::value;
+  bool useAnalyticalResults = xBool && aBool;
+
+#if defined(KOKKOSKERNELS_INST_LAYOUTLEFT) || \
+    (!defined(KOKKOSKERNELS_ETI_ONLY) &&      \
+     !defined(KOKKOSKERNELS_IMPL_CHECK_ETI_CALLS))
+  KOKKOS_IMPL_DO_NOT_USE_PRINTF(
+      "+-----------------------------------------------------------------------"
+      "---\n");
+  KOKKOS_IMPL_DO_NOT_USE_PRINTF("Starting %s for LAYOUTLEFT ...\n",
+                                caseName.c_str());
+
+  if (true) {
+    Test::SyrTester<ScalarX, Kokkos::LayoutLeft, ScalarA, Kokkos::LayoutLeft,
+                    Device>
+        tester;
+    tester.test(0, 0);
+    tester.test(1, 0);
+    tester.test(2, 0);
+    tester.test(13, 0);
+    tester.test(1024, 0);
+
+    if (useAnalyticalResults) {
+      tester.test(1024, 0, true, false, false);
```

Comment:
Consider naming these booleans to improve readability. For example:
`tester.test(1024, 0, useAnalyticalResults, ...)`